vanilla-package/src/test/java/io/jenkins/jenkinsfile/runner/vanilla/SmokeTest.java

```diff
@@ -130,17 +144,84 @@ public void shouldSupportDataboundMethods() throws Throwable {
     public void helloWorldAsYaml() throws Throwable {
         File jenkinsfile = tmp.newFile("Jenkinsfile.yml");
         FileUtils.writeStringToFile(jenkinsfile,
-        "pipeline:\n" +
-        "  agent:\n" +
-        "    none:\n" +
-        "  stages:\n" +
-        "    - stage: \"Print Hello\"\n" +
-        "      steps:\n" +
-        "        - echo \"Hello, world!\""
-        , Charset.defaultCharset());
+                "pipeline:\n" +
+                        "  agent:\n" +
+                        "    none:\n" +
+                        "  stages:\n" +
+                        "    - stage: \"Print Hello\"\n" +
+                        "      steps:\n" +
+                        "        - echo \"Hello, world!\""
+                , Charset.defaultCharset());
 
         int result = JFRTestUtil.runAsCLI(jenkinsfile);
         assertThat("JFR should be executed successfully", result, equalTo(0));
         assertThat(systemOut.getLog(), containsString("Hello, world!"));
     }
+
+    @Test
+    public void checkoutSCM() throws Throwable {
+        Map<String,String> filesAndContents = new HashMap<>();
+        filesAndContents.put("README.md", "Test repository");
+
+        File jenkinsfile = new File(getClass().getResource("SmokeTest/checkoutSCM/Jenkinsfile").getFile());
+        String jfContent = FileUtils.readFileToString(jenkinsfile, Charset.defaultCharset());
+        filesAndContents.put("Jenkinsfile", jfContent);
+
+        String scmConfigPath = createTestRepoWithContentAndSCMConfigYAML(filesAndContents, "master");
+
+        int result = JFRTestUtil.runAsCLI(jenkinsfile, Arrays.asList("--scm", scmConfigPath));
+        assertThat("JFR should be executed successfully", result, equalTo(0));
+        assertThat(systemOut.getLog(), containsString("README.md exists with content 'Test repository'"));
+        assertThat(systemOut.getLog(), containsString("using credential user1"));
+    }
+
+    private String createTestRepoWithContentAndSCMConfigYAML(Map<String,String> filesAndContents, String branch) throws Exception {
+        File gitDir = tmp.newFolder();
+        FilePath gitDirPath = new FilePath(gitDir);
+        GitClient git = Git.with(TaskListener.NULL, new EnvVars()).in(gitDir).getClient();
+        git.init();
+        PersonIdent johnDoe = new PersonIdent("John Doe", "john@doe.com");
+        for (String fileName : filesAndContents.keySet()) {
+            FilePath file = gitDirPath.child(fileName);
+            try {
+                file.write(filesAndContents.get(fileName), null);
+            } catch (Exception e) {
+                throw new GitException("unable to write file", e);
+            }
+            git.add(fileName);
+        }
+        git.setAuthor(johnDoe);
+        git.setCommitter(johnDoe);
+        git.commit("Test commit");
+
+        Map<String,Object> remoteConfig = Stream.of(new Object[][]{
```

Comment:
I don't know if there's a better way to do arbitrary map setup than this - the `toYaml` stuff in JCasC is all `NoExternalUse` so I can't just create the `GitSCM` and credentials and then `toYaml` them without being a bad boy. =)